config/repository_resource.go

```diff
@@ -176,40 +175,50 @@ func zypperRepoContents(repo *agentendpointpb.OSPolicy_Resource_RepositoryResour
 	return buf.Bytes()
 }
 
-func fetchGPGKey(key string) ([]byte, error) {
+func serializeGPGKeyEntity(entityList openpgp.EntityList) ([]byte, error) {
+	var buf bytes.Buffer
+	for _, entity := range entityList {
+		if err := entity.Serialize(&buf); err != nil {
+			return nil, fmt.Errorf("error serializing gpg key: %v", err)
+		}
+	}
+	return buf.Bytes(), nil
+}
+
+func isArmoredGPGKey(keyData []byte) bool {
+	var buf bytes.Buffer
+	tee := io.TeeReader(bytes.NewReader(keyData), &buf)
+
+	// Try decoding as armored
+	decodedBlock, err := armor.Decode(tee)
+	if err == nil && decodedBlock != nil {
+		return true
+	}
+
+	return false
+}
+
+func fetchGPGKey(key string) (openpgp.EntityList, error) {
 	resp, err := http.Get(key)
 	if err != nil {
-		return nil, fmt.Errorf("error downloading gpg key: %v", err)
```

Comment:
The standard approach is to use `%w` for `err` so as to chain the errors (here and below).